### PR TITLE
[eslint] Do not access Object.prototype method 'hasOwnProperty' from target object no-prototype-builtins

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

Solves eslint errors for no-prototype-builtins and useful for removing the off found in this PR: https://github.com/aces/Loris/pull/7137

See https://eslint.org/docs/rules/no-prototype-builtins

### CODE REVIEW

File: jsx/Form.js [hunk 1]
```diff
@@ -370,7 +370,7 @@ class SearchableDropdown extends Component {
     let optionList = [];
     if (sortByValue) {
       for (let key in options) {
-        if (options.hasOwnProperty(key)) {
```
Comment (driusan) [inline jsx/Form.js:373]: Can't these loops just be changed to `for .. of ..` and the hasOwnProperty check be removed?

Comment (laemtl) [reply jsx/Form.js:373]: The for..of loop only supports iterable objects like arrays, not objects. I usually use a combination of for of and Object.entries.

Comment (driusan) [reply jsx/Form.js:373]: That still sounds cleaner than Object.prototype.hasOwnProperty.call

Comment (maltheism) [reply jsx/Form.js:373]: Hi @driusan & @laemtl, please see the latest commit for the change.

Comment (driusan) [reply jsx/Form.js:373]: The latest commit looks like it only updated 1 place, the rest of the lines in the PR are still using "hasOwnProperty"

Comment (maltheism) [reply jsx/Form.js:373]: Last one taking longer to fix because I want to test it and I'm needing to configure couchdb.

Comment (maltheism) [reply jsx/Form.js:373]: @driusan @laemtl okay fixed

**Comment by driusan** [comment]
@laemtl can you review?

**Comment by laemtl** [comment]
@maltheism Let me know once the PR is ready.

**Comment by maltheism** [comment]
Hi @laemtl, this PR needs to be tested wherever the code was changed. I'm probably going to add console.log(typeof <var>); everywhere like I did for the dqt module. Then we can go throughout Loris and see if the console.log messages are correct with the code. Let me know if you have any suggestions.

**Comment by driusan** [comment]
@kongtiaowang can you test this PR for @maltheism ?

**Comment by kongtiaowang** [comment]
Screen Shot 2021-03-31 at 10 43 07 AM

**Comment by maltheism** [comment]
@driusan fixed

**Comment by driusan** [comment]
@maltheism it looks like its failing?

**Comment by maltheism** [comment]
Hi @driusan, I haven't had time to fix this PR because I've been focusing on the DQT performance and it's a priority since the release of CBIGR is soon. I'll make sure to get this PR passing next week.

**Comment by maltheism** [comment]
Hi all (@laemtl, @kongtiaowang, @driusan) this PR is needing a review now that all checks are passing.

**Comment by cmadjar** [comment]
@jstirling91 could you please review this? Thanks!

File: modules/dqt/jsx/react.app.js [hunk 2]
```diff
@@ -127,26 +127,26 @@ class DataQueryApp extends Component {
...
-              dataType: 'json',
-            }).then((value) => {
-            let queries = this.state.savedQueries;
-
-            queries[value._id] = value;
-            this.setState({savedQueries: queries});
-          });
-          promises.push(curRequest);
-        }
+    console.log('this.state.queryIDs is ');
```
Comment (jstirling91) [inline modules/dqt/jsx/react.app.js:130]: There's a lot of `console.logs` added in this file. Please remove all of them

File: modules/dqt/jsx/react.fieldselector.js [hunk 3]
```diff
@@ -401,28 +401,28 @@ class FieldSelector extends Component {
     let isFile;
     let fieldName;
     let category;
-    for (i in this.state.categoryFields[this.state.selectedCategory]) {
-      if (this.state.categoryFields[
+    console.log('this.state.categoryFields[this.state.selectedCategory] is ');
```
Comment (jstirling91) [inline modules/dqt/jsx/react.fieldselector.js:404]: Same for this file, please remove all `console.log`

File: modules/dqt/jsx/react.filterBuilder.js [hunk 4]
```diff
@@ -299,14 +299,14 @@ class FilterRule extends Component {
...
         operatorKey = inputType[0];
-        for (let key in this.state.operators) {
-          if (this.state.operators.hasOwnProperty(key)) {
-            operators.push(
-              <option key={key} value={key} onChange={this.operatorSelect}>
-                {this.state.operators[key]}
-              </option>
-            );
-          }
+        console.log('this.state.operators is ');
```
Comment (jstirling91) [inline modules/dqt/jsx/react.filterBuilder.js:302]: Remove these

File: modules/dqt/jsx/react.tabs.js [hunk 5]
```diff
@@ -459,8 +459,10 @@ class ViewDataTabPane extends Component {
       </>
     );
     let criteria = [];
-    for (let el in this.props.Criteria) {
-      if (!this.props.Criteria.hasOwnProperty(el)) {
+    console.log('this.props.Criteria is ');
```
Comment (jstirling91) [inline modules/dqt/jsx/react.tabs.js:462]: Remove all `console.log` from file

**Review by jstirling91 — CHANGES_REQUESTED** [review]
There's a lot of debug code still left in this PR. Please update and remove all `console.log` added

**Comment by maltheism** [comment]
@jstirling91 ready

**Comment by driusan** [comment]
@maltheism it's still failing

**Comment by driusan** [comment]
I've rebased this but I don't think I'll have time to look into the new errors. We should assign this to someone else. (There were also a bunch of conflicts with the dqt module that I'm not sure were resolved correctly so whoever takes this over should ensure that it's well tested.)

**Comment by driusan** [comment]
I don't think this is going to happen without anyone here to take care of the PR, so I'm going to close this. If someone can take it over it can be re-opened.

**Pull request closed without merging**